data-types/fixed-length-ascii/README.md

```diff
@@ -0,0 +1,33 @@
+# Fixed-length ASCII data type
+
+Defines a data type for fixed-length ASCII strings.
```

Comment:
I think my initial name for this dtype (fixed-length-ascii) was misleading -- it models the numpy `S` dtype, which represents any fixed-length byte string, which includes non-ascii characters. So we should probably rename this to `fixed-length-bytes` or equivalent.

Reply:
i think the key distinction between this dtype (modelling numpy `S*`) and the fixed-length bytes dtype defined elsewhere in this PR (modelling numpy `V*`) is that scalars with this dtype are intended to be interpreted as strings, not opaque sequences of bytes. We should probably decide if we really want two dtypes for this.

Reply:
Numpy also assumes NUL-padding, i.e. that any NUL characters at the end can be ignored.